tests/utils/siteBuilder.js

```diff
@@ -22,74 +33,75 @@ const createSiteBuilder = ({ siteName }) => {
       const dest = path.join(directory, pathPrefix, 'netlify.toml')
       const content = toToml(config, { space: 2 })
       tasks.push(async () => {
-        await fs.ensureFile(dest)
+        await ensureDir(path.dirname(dest))
         await fs.writeFile(dest, content)
       })
       return builder
     },
     withPackageJson: ({ object, pathPrefix = '' }) => {
       const dest = path.join(directory, pathPrefix, 'package.json')
       tasks.push(async () => {
-        await fs.ensureFile(dest)
-        await fs.writeJSON(dest, object, { spaces: 2 })
+        const content = JSON.stringify(object, null, 2)
+        await ensureDir(path.dirname(dest))
+        await fs.writeFile(dest, `${content}\n`)
       })
       return builder
     },
     withFunction: ({ pathPrefix = 'functions', path: filePath, handler }) => {
       const dest = path.join(directory, pathPrefix, filePath)
       tasks.push(async () => {
-        await fs.ensureFile(dest)
+        await ensureDir(path.dirname(dest))
         await fs.writeFile(dest, `exports.handler = ${handler.toString()}`)
       })
       return builder
     },
     withEdgeHandlers: ({ handlers }) => {
       const dest = path.join(directory, 'edge-handlers', 'index.js')
       tasks.push(async () => {
-        await fs.ensureFile(dest)
         const content = Object.entries(handlers)
           .map(([event, handler]) => {
             return `export const ${event} = ${handler.toString()}`
           })
           .join('\n')
```

Comment:
Unrelated, but this is the only join not using `os.EOL`.

Reply:
This file is processed only by our infrastructure at the moment (not also locally like other files).